src/core/client/auth/components/SignUp.tsx

```diff
@@ -0,0 +1,198 @@
+import * as React from "react";
+import { StatelessComponent } from "react";
+import { Field, Form } from "react-final-form";
+import { OnSubmit } from "talk-framework/lib/form";
+import {
+  composeValidators,
+  required,
+  validateEmail,
+  validateEqualPasswords,
+  validatePassword,
+  validateUsername,
+} from "talk-framework/lib/validation";
+import {
+  Button,
+  CallOut,
+  Flex,
+  FormField,
+  InputDescription,
+  InputLabel,
+  TextField,
+  Typography,
+  ValidationMessage,
+} from "talk-ui/components";
+import { View } from "../containers/SignUpContainer";
+import * as styles from "./SignUp.css";
+
+interface FormProps {
+  email: string;
+  username: string;
+  password: string;
+  confirmPassword: string;
+}
+
+export interface SignUpForm {
+  onSubmit: OnSubmit<FormProps>;
+  setView: (view: View) => void;
```

Comment:
Should be `onSignIn` or similar to avoid putting logic in this presentational component.